build-llvm.py

```diff
@@ -337,60 +365,69 @@ def invoke_cmake(build, cc, cxx, debug, install_folder, ld, projects, root,
         defines['LLVM_CCACHE_BUILD'] = 'ON'
 
     # If we found a linker, we should use it
-    if ld is not None:
-        defines['LLVM_USE_LINKER'] = ld
+    if env_vars.ld is not None:
+        defines['LLVM_USE_LINKER'] = env_vars.ld
 
     # Add the defines, point them to our build folder, and invoke cmake
     cmake = ['cmake', '-G', 'Ninja', '-Wno-dev']
     for key in defines:
         newdef = '-D' + key + '=' + defines[key]
         cmake += [newdef]
-    cmake += [root.joinpath("llvm-project", "llvm").as_posix()]
+    cmake += [root_folder.joinpath("llvm-project", "llvm").as_posix()]
 
-    subprocess.run(cmake, check=True, cwd=build.as_posix())
+    subprocess.run(cmake, check=True, cwd=build_folder.as_posix())
 
 
-# Build the world
-def invoke_ninja(build, install_folder):
-    utils.header("Building LLVM")
+def invoke_ninja(dirs):
+    """
+    Invoke ninja to run the actual build
+    :param dirs: An instance of the Directories class with the paths to use
+    :return:
+    """
+    utils.print_header("Building LLVM")
 
-    timeStarted = time.time()
+    build_folder = dirs.build_folder
+    install_folder = dirs.install_folder
 
-    subprocess.run('ninja', check=True, cwd=build.as_posix())
+    time_started = time.time()
+
+    subprocess.run('ninja', check=True, cwd=build_folder.as_posix())
 
     print()
     print("LLVM build duration: " +
-          str(datetime.timedelta(seconds=int(time.time() - timeStarted))))
+          str(datetime.timedelta(seconds=int(time.time() - time_started))))
 
     subprocess.run(['ninja', 'install'],
                    check=True,
-                   cwd=build.as_posix(),
+                   cwd=build_folder.as_posix(),
                    stdout=subprocess.DEVNULL,
                    stderr=subprocess.DEVNULL)
 
     utils.create_gitignore(install_folder)
 
 
 def main():
-    root = pathlib.Path(__file__).resolve().parent
-    build = root.joinpath("build", "llvm")
+    root_folder = pathlib.Path(__file__).resolve().parent
+    build_folder = root_folder.joinpath("build", "llvm")
 
-    args = parse_parameters(root)
+    args = parse_parameters(root_folder)
 
     install_folder = pathlib.Path(args.install_folder)
     if not install_folder.is_absolute():
         install_folder = root.joinpath(install_folder)
 
+    dirs = Directories(build_folder, install_folder, root_folder)
+
     cc, cxx, ld = check_cc_ld_variables()
+    env_vars = EnvVars(cc, cxx, ld)
```

Comment:
I *think* this could be `env_vars = EnvVars(*check_cc_ld_variables())`: https://docs.python.org/3/tutorial/controlflow.html#unpacking-argument-lists